lib/clients/websocket.js

```diff
@@ -17,12 +17,12 @@ class WebsocketClient extends EventEmitter {
     { channels = null } = {}
   ) {
     super();
-    this.productIDs = Utils.determineProductIDs(productIDs);
+    this.productIDs = new Set(Utils.determineProductIDs(productIDs));
     this.websocketURI = websocketURI;
     this.auth = Utils.checkAuth(auth);
-    this.channels = channels || ['full'];
-    if (!this.channels.includes('heartbeat')) {
-      this.channels.push('heartbeat');
+    this.channels = new Set(channels || ['full']);
+    if (!this.channels.has('heartbeat')) {
+      this.channels.add('heartbeat');
     }
```

Comment:
Good use of `Set` here, but there's no need to check if `channels` already has `heartbeat` before adding it. That's the benefit of using the `Set`. 😃